examples/tensorflow/image-classifier-resnet50/README.md

```diff
@@ -0,0 +1,96 @@
+
+1. [cortex_inf.yaml](cortex_inf.yaml) - to be used with `inf1` instances.
+1. [cortex_cpu.yaml](cortex_cpu.yaml) - to be used with any instances that have CPUs.
+1. [cortex_gpu.yaml](cortex_gpu.yaml) - to be used with GPU instances.
+
+To deploy an API, run:
+
+```bash
+cortex deploy <cortex-deployment-yaml>
+```
+
+E.g.
+
+```bash
+cortex deploy cortex_cpu.yaml
+```
+
+## Verifying your API
+
+Check that your API is live by running `cortex get image-classifier-resnet50`, and copy the example `curl` command that's shown. After the API is live, run the `curl` command, e.g.
+
+```bash
+curl <API endpoint> -X POST -H "Content-Type: application/json" -d @sample.json
+```
+
+This image is embedded in [sample.json](sample.json):
+
+![image](https://i.imgur.com/213xcvs.jpg)
+
+If a 5-element list is returned containing classifications of the image ("tabby", "Egyptian_cat", "tiger_cat", "tiger", "plastic_bag"), then it means the API is working. The first classification in the list is the most likely.
+
+## Throughput test
+
+[throughput_test.py](throughput_test.py) is a Python CLI that can be used to test the throughput of your deployed API. The throughput will vary depending on your API's configuration (specified in the `cortex_*.yaml` file), your local machine's resources (mostly CPU, since it has to spawn many concurrent requests), and the internet connection on your local machine.
+
+```bash
+Usage: throughput_test.py [OPTIONS] IMG_URL ENDPOINT
+
+  Program for testing the throughput of a Resnet50 model on instances equipped
+  with CPU, GPU or Inferentia devices.
+
+Options:
+  -w, --workers INTEGER     Number of workers for prediction requests.  [default: 1]
+  -t, --threads INTEGER     Number of threads per worker for prediction requests.  [default: 1]
+  -s, --samples INTEGER     Number of samples to run per thread.  [default: 10]
+  -i, --time-based FLOAT    How long the thread making predictions will run for in seconds.
+                            If set, -s option will be ignored.
+  -b, --batch-size INTEGER  Number of images sent for inference in one request.  [default: 1]
+  --help                    Show this message and exit.
+```
+
+The Python CLI has been tested with Python 3.6.9. To install the CLI's dependencies, run the following:
+
+```bash
+pip install requests click opencv-contrib-python numpy
+```
+
+Before [throughput_test.py](throughput_test.py) is run, 2 environment variables have to be exported:
+
+```bash
+export ENDPOINT=<API endpoint>  # you can find this with `cortex get image-classifier-resnet50`
+export IMG_URL=https://i.imgur.com/213xcvs.jpg # this is the cat image shown in the previous step
+```
+
+Then, deploy each API one at a time and check the results:
+
+1. Running `python throughput_test.py -i 30 -w 4 -t 48` with the [cortex_inf.yaml](cortex_inf.yaml) API running on an `inf1.2xlarge` instance will get **~510 inferences/sec** with an average latency of **80 ms**.
```

Comment:
We want to show how much performance you can get per dollar with Inferentia ASICs. The only instance that makes sense economically is `inf1.2xlarge`, which is almost twice as cost-effective as `inf1.xlarge`. That's because the major bottleneck is with the CPU and not with the Inferentia ASIC. If the CPU were to be more performant, the throughput could theoretically be as almost as 8 times as higher (at least in this example).

Basically, when the throughput is high, the costs with `inf1.xlarge` are high too, but with `inf1.2xlarge` they are low.

Reply:
Add reason why `inf1.xlarge` wasn't used in this example.